vectormaster/src/main/java/com/sdsmdg/harjot/vectormaster/models/GroupModel.java

```diff
@@ -158,7 +158,7 @@ public PathModel getPathModelByName(String name) {
     public ClipPathModel getClipPathModelByName(String name) {
         ClipPathModel cModel = null;
         for (ClipPathModel clipPathModel : getClipPathModels()) {
-            if (clipPathModel.getName().equals(name)) {
+            if (clipPathModel.getName() != null && clipPathModel.getName().equals(name)) {
```

Comment:
`clipPathModel != null && Objects.equals(clipPathModel.getName(), name)` - might be safer/easier

Reply:
How is `Objects.equals` better ?

Reply:
If `getName()` returns `null` then the `.equals()` will trigger a NPE. The way I have suggested would not suffer from this.

Reply:
If `getName()` returns `null`, the `equals()` condition would not be checked.

Reply:
This is true, I was commenting based on what I suggested. `clipPathModel` could also be null so I would suggest checking that. Another suggestion is to not allow `getName()` to be null at all (it could still be empty) at the model level.

Reply:
Also based on your code as it stands you would not have to check if `getName()` is null using `Objects.equals(clipPathModel.getName(), name)` - if it returns null then `equals` will return false. Hopefully this shows some benefit.